lib/import/colorectal/providers/st_george/st_george_handler_colorectal.rb

```diff
@@ -0,0 +1,317 @@
+require 'possibly'
+require 'pry'
+require 'date'
+
+module Import
+  module Colorectal
+    module Providers
+      module StGeorge
+        # Process St George-specific record details into generalized internal genotype format
+        class StGeorgeHandlerColorectal < Import::Germline::ProviderHandler
+          include Import::Helpers::Colorectal::Providers::Rj7::Constants
+
+          def process_fields(record)
+            genotype = Import::Colorectal::Core::Genocolorectal.new(record)
+
+            # records using new importer should only have SRIs starting with V
+            return unless record.raw_fields['servicereportidentifier'].start_with?('V')
+
+            # add standard passthrough fields to genotype object
+            genotype.add_passthrough_fields(record.mapped_fields, record.raw_fields,
+                                            PASS_THROUGH_FIELDS)
+
+            assign_test_type(genotype, record)
+            genotype = assign_test_scope(genotype, record)
+            genotypes = fill_genotypes(genotype, record)
+
+            genotypes.each do |single_genotype|
+              process_variants(single_genotype, record)
+
+              @persister.integrate_and_store(single_genotype)
+            end
+          end
+
+          def assign_test_type(genotype, record)
+            # extract molecular testing type from the raw record
+            # map molecular testing type and assign to genotype using
+            # add_molecular_testing_type_strict method from genocolorectal.rb
+
+            return if record.raw_fields['moleculartestingtype'].blank?
+
+            return unless TEST_TYPE_MAP[record.raw_fields['moleculartestingtype']&.downcase&.strip]
+
+            genotype.add_molecular_testing_type_strict(TEST_TYPE_MAP[record.raw_fields['moleculartestingtype']&.downcase&.strip])
+          end
+
+          def assign_test_scope(genotype, record)
+            # extract molecular testing type from the raw record
+            # map molecular testing type and assign to genotype using
+            # add_test_scope method from genocolorectal.rb
+            # return (filled) genotype
+
+            testscope = record.raw_fields['moleculartestingtype']&.downcase&.strip
+            genotype.add_test_scope(TEST_SCOPE_MAP[testscope])
+            return genotype if genotype.attribute_map['genetictestscope'].present?
+
+            genotype.add_test_scope(:no_genetictestscope)
+
+            genotype
```

Comment:
You won't need to return the `genotype` here if `process_fields` doesn't reassign `genotype`.

Reply:
removed in  [b64fd4c](https://github.com/NHSDigital/data_management_system/commit/b64fd4c1b851661d098425a1ab1df681db92063f)

Reply:
removed